src/hooks/useOnClickOutside.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
+
+export const useOnClickOutside = (
+  ref: React.RefObject<HTMLElement>,
+  cb: () => void
+) => {
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (ref.current && !ref.current.contains(event.target)) {
+        cb();
+      }
+    }
+    // Bind the event listener
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
```

Comment:
TLDR: Comments do not add value for code where it is clear from reading react docs / having FE knowledge what it does.

i appreciate you adding more comments! Sarah Drasner made a great tweet about comments which is what inspired me to write more of them. What she said is that comments should explain the "why" about a code. For example, regex should always have a code because no one can read it. Complex or unclear code should have a comment explaining why it's there so that a future coder knows not to remove it. Or so they know how to refactor it. 

Comments like `Bind the event listener` and `Unbind the event listener on clean up` however basically restate the code they describe. They answer the "what" instead of the "why". Therefore the reader has to read double: the code and the comment which both convey the same thing.